internal/cli/command/cluster/templates.go

```diff
@@ -0,0 +1,202 @@
+				},
+			},
+			Kubernetes: pipeline.CreatePkeClusterKubernetes{
+				Version: "1.17.9",
+				Rbac:    true,
+			},
+		},
+		pkeOnVsphere: pipeline.CreatePkeOnVsphereClusterRequest{
+			Type: pkeOnVsphere,
+			Kubernetes: pipeline.CreatePkeClusterKubernetes{
+				Version: "1.15.3",
+				Rbac:    true,
+			},
+			Folder:       "folder",
+			Datastore:    "DatastoreCluster",
+			ResourcePool: "resource-pool",
+			Nodepools: []pipeline.PkeOnVsphereNodePool{
+				{
+					Name:          "master",
+					Roles:         []string{"master", "worker"},
+					Size:          1,
+					Vcpu:          2,
+					Ram:           1024,
+					Template:      "pke-template",
+					AdminUsername: "root",
+				},
+			},
+		},
+		"ack": pipeline.CreateClusterRequest{
+			Cloud: "alibaba",
+			Properties: map[string]interface{}{
+				"ack": pipeline.CreateAckPropertiesAck{},
+			},
+		},
+		"aks": pipeline.CreateClusterRequest{
+			Cloud:    "azure",
+			Location: "westus2",
+			Properties: map[string]interface{}{
+				"aks": pipeline.CreateAksPropertiesAks{
+					KubernetesVersion: "1.17.11",
+					NodePools: map[string]pipeline.NodePoolsAzure{
+						"pool1": {
+							Count:        3,
+							Autoscaling:  true,
+							InstanceType: "Standard_B2s",
+							MinCount:     3,
+							MaxCount:     4,
+						},
+					},
+				},
+			},
+		},
+		"eks": pipeline.CreateClusterRequest{
+			Cloud:    "amazon",
+			Location: "us-east-2",
+			Properties: map[string]interface{}{
+				"eks": pipeline.CreateEksPropertiesEks{
+					Version: "1.16.13",
+					NodePools: map[string]pipeline.EksNodePool{
+						"pool1": {
+							Count:        3,
+							MinCount:     3,
+							MaxCount:     4,
+							Autoscaling:  true,
+							InstanceType: "t2.medium",
+						},
+					},
+				},
+			},
+		},
+		"gke": pipeline.CreateClusterRequest{
+			Cloud:    "google",
+			Location: "us-west1-b",
+			Properties: map[string]interface{}{
+				"gke": pipeline.CreateGkePropertiesGke{
+					Master: pipeline.CreateGkePropertiesGkeMaster{
+						Version: "1.15.12-gke.20",
+					},
+					NodeVersion: "1.15.12-gke.20",
+					NodePools: map[string]pipeline.NodePoolsGoogle{
+						"pool1": {
+							Count: 3, Autoscaling: true,
+							InstanceType: "e2-medium",
+							Preemptible:  true,
+							MinCount:     3,
+							MaxCount:     4,
+						},
+					},
+				},
+			},
+		},
+		"oke": pipeline.CreateClusterRequest{
+			Cloud:    "oracle",
+			Location: "eu-frankfurt-1",
+			Properties: map[string]interface{}{
+				"oke": pipeline.CreateUpdateOkePropertiesOke{
+					Version: "v1.14.8",
+					NodePools: map[string]pipeline.NodePoolsOracle{
+						"pool1": {
+							Count:   3,
+							Shape:   "VM.Standard.E2.2",
+							Version: "v1.14.8",
+						},
+					},
+				},
+			},
+		},
+	}
+)
+
+func convertCreateTemplate(in interface{}) (map[string]interface{}, error) {
+	var out map[string]interface{}
+	b, _ := json.Marshal(in)
+	if err := json.Unmarshal(b, &out); err != nil {
```

Comment:
**Question**: wouldn't this be easier with `mapstructure.Decode()` to decode the strongly typed object into a `map[string]interface{}`? Am I missing something here?
Even without explicit struct tags, I believe it would give the same result as the JSON marshalling.

It's not a big deal, just being curios as I feel the actual JSON representation has no purpose here, we are only using it for strong->weak structural transformation, which I believe `mapstructure` is more straightforward for.

Reply:
Unfortunately we cannot use `mapstructure.Decode()` here because the generated structs (like `CreateClusterRequest`) don't have `mapstructure` tag keys. This breaks the `build request` flow (we try to find out the `cloud`, `name`, `type` etc.. field [like this](https://github.com/banzaicloud/banzai-cli/blob/master/internal/cli/command/cluster/create.go#L326))

We can rewrite this method but I think we should do this in another PR

Reply:
(The tag names in mapstructure can be configured, so you can use `json` tags as well: https://github.com/mitchellh/mapstructure/blob/v1.3.3/mapstructure.go#L241)

Reply:
nice, thx @bonifaido, I changed this method to using mapstructure